From 3218ea36c229f0bbdbbe0e4e3ddcee6c68c86691 Mon Sep 17 00:00:00 2001
From: Philippe Normand <philn@igalia.com>
Date: Thu, 20 Mar 2025 05:43:30 -0700
Subject: [PATCH] [GTK][WPE] Fix clang-20 warnings
 https://bugs.webkit.org/show_bug.cgi?id=290093

Reviewed by Michael Catanzaro.

Fix unused-private-field warnings raised by clang-20.

* Source/WebCore/Modules/webcodecs/WebCodecsAudioEncoder.h:
* Source/WebCore/html/MediaElementSession.h:
* Source/WebKit/UIProcess/API/APIProcessPoolConfiguration.h:

Canonical link: https://commits.webkit.org/292414@main
---
 Source/WebCore/Modules/webcodecs/WebCodecsAudioEncoder.h      | 1 -
 Source/WebCore/html/MediaElementSession.h                     | 4 ----
 Source/WebKit/UIProcess/API/APIProcessPoolConfiguration.h     | 2 +-
 .../WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h    | 3 ---
 4 files changed, 1 insertion(+), 9 deletions(-)

diff --git a/Source/WebCore/Modules/webcodecs/WebCodecsAudioEncoder.h b/Source/WebCore/Modules/webcodecs/WebCodecsAudioEncoder.h
index bbd805844c75..a3259783a364 100644
--- a/Source/WebCore/Modules/webcodecs/WebCodecsAudioEncoder.h
+++ b/Source/WebCore/Modules/webcodecs/WebCodecsAudioEncoder.h
@@ -82,7 +82,6 @@ private:
 
     WebCodecsEncodedAudioChunkMetadata createEncodedChunkMetadata();
 
-    size_t m_encodeQueueSize { 0 };
     Ref<WebCodecsEncodedAudioChunkOutputCallback> m_output;
     Ref<WebCodecsErrorCallback> m_error;
     RefPtr<AudioEncoder> m_internalEncoder;
diff --git a/Source/WebCore/html/MediaElementSession.h b/Source/WebCore/html/MediaElementSession.h
index 1836e96fa4fe..f70130035ad1 100644
--- a/Source/WebCore/html/MediaElementSession.h
+++ b/Source/WebCore/html/MediaElementSession.h
@@ -245,10 +245,6 @@ private:
     Timer m_mainContentCheckTimer;
     Timer m_clientDataBufferingTimer;
 
-#if !RELEASE_LOG_DISABLED
-    uint64_t m_logIdentifier { 0 };
-#endif
-
 #if ENABLE(MEDIA_USAGE)
     bool m_haveAddedMediaUsageManagerSession { false };
 #endif
diff --git a/Source/WebKit/UIProcess/API/APIProcessPoolConfiguration.h b/Source/WebKit/UIProcess/API/APIProcessPoolConfiguration.h
index 26bd3031d2f9..af4944e5a5d3 100644
--- a/Source/WebKit/UIProcess/API/APIProcessPoolConfiguration.h
+++ b/Source/WebKit/UIProcess/API/APIProcessPoolConfiguration.h
@@ -193,7 +193,7 @@ private:
     bool m_shouldConfigureJSCForTesting { false };
     bool m_isJITEnabled { true };
     bool m_usesSingleWebProcess { false };
-#if PLATFORM(GTK) && !USE(GTK4)
+#if PLATFORM(GTK) && !USE(GTK4) && USE(CAIRO)
     bool m_useSystemAppearanceForScrollbars { false };
 #endif
 #if PLATFORM(PLAYSTATION)
diff --git a/Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h b/Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h
index f19473587ec9..3aa8a0a30ba5 100644
--- a/Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h
+++ b/Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h
@@ -184,9 +184,7 @@ private:
     WebCore::GraphicsLayer* m_rootCompositingLayer { nullptr };
     WebCore::GraphicsLayer* m_overlayCompositingLayer { nullptr };
     HashSet<Ref<WebCore::CoordinatedPlatformLayer>> m_layers;
-    bool m_didInitializeRootCompositingLayer { false };
     bool m_layerTreeStateIsFrozen { false };
-    bool m_isPurgingBackingStores { false };
     bool m_pendingResize { false };
     bool m_pendingForceRepaint { false };
     bool m_isSuspended { false };
@@ -197,7 +195,6 @@ private:
 #if ENABLE(SCROLLING_THREAD)
     bool m_compositionRequiredInScrollingThread { false };
 #endif
-    double m_lastAnimationServiceTime { 0 };
     RefPtr<ThreadedCompositor> m_compositor;
     struct {
         CompletionHandler<void()> callback;
-- 
2.48.1

